quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -614,15 +589,19 @@ void disableHealthEndpoint(BuildProducer<RouteBuildItem> routes, BuildProducer<B
     }
 
     @BuildStep
-    void configureResteasy(BuildProducer<ResteasyDeploymentCustomizerBuildItem> deploymentCustomizerProducer) {
-        deploymentCustomizerProducer.produce(new ResteasyDeploymentCustomizerBuildItem(new Consumer<ResteasyDeployment>() {
+    void configureResteasy(CombinedIndexBuildItem index,
+            BuildProducer<BuildTimeConditionBuildItem> buildTimeConditionBuildItemBuildProducer,
+            BuildProducer<MethodScannerBuildItem> scanner) {
+        buildTimeConditionBuildItemBuildProducer.produce(new BuildTimeConditionBuildItem(index.getIndex().getClassByName(DotName.createSimple(
+                KeycloakApplication.class.getName())), false));
```

Comment:
Just out of curiosity, what exactly is this change for? What does the `BuildTimeConditionBuildItem` do? I haven't found any docs for it.

Reply:
Removes the given type from Arc index. As we don't want `KeycloakApplication` available we disable it.